src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -370,17 +370,17 @@ private static void finish(@CheckForNull String cookie) {
                 Timer.get().submit(new Runnable() { // JENKINS-31614
                     @Override public void run() {
                         execution.completed(null);
+                        try {
+                            runningTask.launcher.kill(Collections.singletonMap(COOKIE_VAR, cookie));
+                        } catch (ChannelClosedException x) {
+                            // fine, Jenkins was shutting down
+                        } catch (RequestAbortedException x) {
+                            // slave was exiting; too late to kill subprocesses
```

Comment:
Could perhaps log something at `FINE`; it is not considered a problem.

BTW this code is not new, merely reindented.